src/Microsoft.AspNet.Razor/TagHelpers/TagHelperDescriptorMatcher.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Framework.Internal;
+
+namespace Microsoft.AspNet.Razor.TagHelpers
+{
+    /// <summary>
+    /// Helper methods related to <see cref="TagHelperDescriptor"/> and related classes.
+    /// </summary>
+    public static class TagHelperDescriptorMatcher
```

Comment:
Nah. This class matches names against `TagHelperDescriptor`s.  It would be called `TagHelperDescriptorComparer` if it were a comparer.